### PR TITLE
Support booting from a custom private network.

### PR DESCRIPTION
**Pull request by alboyadjian**
Need some help figuring out better naming for the config options.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,11 +57,19 @@ for your specified platform. Additional, optional overrides can be provided:
...
     no_ssh_tcp_check_sleep: [NUM OF SECONDS TO SLEEP IF no_ssh_tcp_check IS SET]
-    networks: [LIST OF RACKSPACE NETWORK UUIDS, DEFAULT PUBLICNET AND SERVICE NET]
+    additional_networks: [LIST OF RACKSPACE NETWORK UUIDS, BESIDES PUBLICNET AND SERVICE NET]
     rackconnect_wait: ['true' IF USING RACKCONNECT TO WAIT FOR IT TO COMPLETE]
     servicelevel_wait: ['true' IF USING MANAGED SERVICE LEVEL AUTOMATION TO WAIT FOR IT TO COMPLETE]
     no_passwd_lock: ['true' IF FOG LIBRARY SHOULD NOT LOCK ROOT ACCOUNT]
-    servicenet: ['true' IF USING THE SERVICENET IP ADDRESS TO CONNECT]
+    servicenet: ['true' IF THE SERVICENET IP ADDRESS SHOULD BE USED TO CONNECT]
+    ssh_network_name: [IF SET, WILL USE THE SPECIFIED NETWORK TO CONNECT]
+    connect_public_net: [DEFAULTS TO true, SET TO false TO PROVISION NO PUBLIC IP]
```
Comment (davexunit) [inline README.md:66]: Maybe `publicnet` to match `servicenet`?

File: README.md [hunk 2]
```diff
@@ -57,11 +57,19 @@ for your specified platform. Additional, optional overrides can be provided:
...
     no_ssh_tcp_check: [DEFAULTS TO false, SKIPS TCP CHECK WHEN true]
     no_ssh_tcp_check_sleep: [NUM OF SECONDS TO SLEEP IF no_ssh_tcp_check IS SET]
-    networks: [LIST OF RACKSPACE NETWORK UUIDS, DEFAULT PUBLICNET AND SERVICE NET]
+    additional_networks: [LIST OF RACKSPACE NETWORK UUIDS, BESIDES PUBLICNET AND SERVICE NET]
     rackconnect_wait: ['true' IF USING RACKCONNECT TO WAIT FOR IT TO COMPLETE]
     servicelevel_wait: ['true' IF USING MANAGED SERVICE LEVEL AUTOMATION TO WAIT FOR IT TO COMPLETE]
     no_passwd_lock: ['true' IF FOG LIBRARY SHOULD NOT LOCK ROOT ACCOUNT]
-    servicenet: ['true' IF USING THE SERVICENET IP ADDRESS TO CONNECT]
+    servicenet: ['true' IF THE SERVICENET IP ADDRESS SHOULD BE USED TO CONNECT]
+    ssh_network_name: [IF SET, WILL USE THE SPECIFIED NETWORK TO CONNECT]
```
Comment (davexunit) [inline README.md:65]: Maybe just `ssh_network`?

File: lib/kitchen/driver/rackspace.rb [hunk 3]
```diff
@@ -168,19 +184,34 @@ def servicelevel_check(server)
...
+
       def networks
-        base_nets = %w(
-          00000000-0000-0000-0000-000000000000
-          11111111-1111-1111-1111-111111111111
-        )
-        config[:networks] ? base_nets + config[:networks] : nil
+        # The .compact call gets rid of nil public_network or
+        # additional_networks.
+        [service_network, public_network, additional_networks].flatten.compact
```
Comment (davexunit) [inline lib/kitchen/driver/rackspace.rb:214]: I think `[service_network, public_network].concat(additional_networks).compact` is a bit more straightforward.

File: README.md (hunk 1 above)
Comment (alboyadjian) [reply README.md:66]: I started with that, but they're not equivalent.  if servicenet is true, then it uses the service net as the ssh network. If servicenet is false, it uses the publicnet (or the override ssh_network if one is specified) but it still connects the servicenet.  On the other hand, if publicnet is false, it doesn't even connect the public network.

File: lib/kitchen/driver/rackspace.rb (hunk 3 above)
Comment (alboyadjian) [reply lib/kitchen/driver/rackspace.rb:214]: yep.  thanks. that's much better.

File: lib/kitchen/driver/rackspace.rb [hunk 4]
```diff
@@ -168,19 +184,34 @@ def servicelevel_check(server)
...
           server.private_ip_address
         end
       end
 
+      private def public_network
+        # Returns nil if no public ip address should be provisioned
+        '00000000-0000-0000-0000-000000000000' if config[:connect_public_net]
+      end
+
+      private def service_network
```
Comment (davexunit) [inline lib/kitchen/driver/rackspace.rb:201]: I thought this was known as the "private" network?  Is "service" a synonym in Rackspace terms?

Comment (alboyadjian) [reply lib/kitchen/driver/rackspace.rb:201]: The "Service" net is a Rackspace-specific term.  It means different things in Rackconnect v2 and v3.  Service net and private network are not exact equivalents.  

I am pulling this out of memory, so I could be wrong, but I believe in RCv2, the network on the 10.x.x.x space served by default as a private network among all the instances within the cloud account.  It also served as the "service" net that allowed rackspace automation service and rackconnect services to administer the boxes.  Things like cloud backup and iptables configuration.   In RCv3, the 10-net is no longer a private network among the cloud boxes (at least by default, as far as I know).  It serves solely as a network for Rackspace management services, and if you want a private network you have to add one.

**Pull request closed without merging**